apis/python/src/tiledbsoma/dense_nd_array.py

```diff
@@ -148,16 +151,64 @@ def read_tensor(
         """
         Read a user-defined dense slice of the array and return as an Arrow ``Tensor``.
         """
-        tiledb_result_order = tiledb_result_order_from_soma_result_order(
-            result_order, accept=["column-major", "row-major"]
-        )
         with self._tiledb_open("r") as A:
             target_shape = _dense_index_to_shape(coords, A.shape, result_order)
-            query = A.query(return_arrow=True, order=tiledb_result_order)
-            arrow_tbl = query.df[coords]
-            return pa.Tensor.from_numpy(
-                arrow_tbl.column("soma_data").to_numpy().reshape(target_shape)
-            )
+            schema = A.schema
+
+        sr = clib.SOMAReader(
+            self._uri,
+            name=self.__class__.__name__,
+            result_order=result_order,
+            platform_config={} if self._ctx is None else self._ctx.config().dict(),
+        )
+
+        if coords is not None:
+            if not isinstance(coords, (list, tuple)):
+                raise TypeError(
+                    f"coords type {type(coords)} unsupported; expected list or tuple"
+                )
+            if len(coords) < 1 or len(coords) > schema.domain.ndim:
+                raise ValueError(
+                    f"coords {coords} must have length between 1 and ndim ({schema.domain.ndim}); got {len(coords)}"
+                )
+
+            for i, coord in enumerate(coords):
+                #                # Example: coords = [None, 3, slice(4,5)]
```

Comment:
Tiny nit (OK to ignore): these types of comments make the code hard to read as they lack context.  They are useful though, as end-user docstrings, so I suggest a nice, clean and thorough docstring at the top of the method, which includes examples such as these.